node/sync/src/block_sync.rs

```diff
@@ -270,12 +277,44 @@ impl<N: Network> BlockSync<N> {
 
         // Construct the message.
         let message = C::prepare_block_request(start_height, end_height);
+
         // Send the message to the peers.
+        let mut tasks = Vec::with_capacity(sync_ips.len());
         for sync_ip in sync_ips {
             let sender = communication.send(sync_ip, message.clone()).await;
-            // If the send fails for any peer, remove the block request from the sync pool.
-            if sender.is_none() {
-                warn!("Failed to send block request to peer '{sync_ip}'");
+            let task = tokio::spawn(async move {
+                // Ensure the request is sent successfully.
+                match sender {
+                    Some(sender) => {
+                        if let Err(err) = sender.await {
+                            warn!("Failed to send block request to peer '{sync_ip}': {err}");
+                            false
+                        } else {
+                            true
+                        }
+                    }
+                    None => {
+                        warn!("Failed to send block request to peer '{sync_ip}': no such peer");
+                        false
+                    }
+                }
+            });
+
+            tasks.push(task);
+        }
+
+        // Wait for all sends to finish at the same time.
```

Comment:
I recall from old benchmarks that we started processing blocks while new requests were still coming in. Is this behaviour changed in this PR or was it already synchronous before?